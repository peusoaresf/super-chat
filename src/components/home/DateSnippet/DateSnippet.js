import React from 'react';
import { Text } from 'react-native';

const DateSnippet = ({ date }) => (
  <Text style={{ fontSize: 11, marginBottom: 10 }}>
    {date}
  </Text>
);

export default DateSnippet;