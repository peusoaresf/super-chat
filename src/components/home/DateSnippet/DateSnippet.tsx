import React from 'react'
import { Text } from 'react-native'

interface DateSnippetProps {
  date: string
}

const DateSnippet = ({ date }: DateSnippetProps) => (
  <Text style={{ fontSize: 11, marginBottom: 10 }}>
    {date}
  </Text>
)

export default DateSnippet
