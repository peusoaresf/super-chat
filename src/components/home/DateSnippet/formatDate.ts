const formatDate = (dateObj: Date) => {
  const isToday = (d: Date) => d.toDateString() === (new Date()).toDateString()

  const isYesterday = (d: Date) => {
    const yest = new Date()
    yest.setDate(yest.getDate() - 1)
    return d.toDateString() === yest.toDateString()
  }

  const toTodayString = (d: Date) => `${d.getHours().toString().padStart(2, '0')}:${d.getMinutes().toString().padStart(2, '0')}`

  if (isToday(dateObj)) {
    return toTodayString(dateObj)
  }

  if (isYesterday(dateObj)) {
    return 'yesterday'
  }

  return dateObj.toLocaleDateString()
}

export default formatDate
