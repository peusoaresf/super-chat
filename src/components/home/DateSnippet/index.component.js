import React from 'react';
import { Text } from 'react-native';

const Component = ({ date, formatDate }) => (
  <Text style={{ fontSize: 11, marginBottom: 10 }}>
    {formatDate(date)}
  </Text>
);

export default Component;