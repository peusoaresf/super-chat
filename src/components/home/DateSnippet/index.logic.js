const formatDate = (dateObj) => {
  const isToday = (d) => d.toDateString() === (new Date()).toDateString();

  const isYesterday = (d) => {
    let yest = new Date();
    yest.setDate(yest.getDate() - 1);
    return d.toDateString() === yest.toDateString();
  }

  const toTodayString = (d) => d.getHours().toString().padStart(2, '0') + ':' + d.getMinutes().toString().padStart(2, '0');

  if (isToday(dateObj)) {
    return toTodayString(dateObj);
  }

  if (isYesterday(dateObj)) {
    return 'yesterday';
  }

  return dateObj.toLocaleDateString();
};

export {
  formatDate
};