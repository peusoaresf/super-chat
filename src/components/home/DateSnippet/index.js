import React from 'react';
import DateSnippet from './DateSnippet';
import { formatDate } from './logic';

const Container = ({ date }) => (
  <DateSnippet date={formatDate(date)} />
);

export default Container;