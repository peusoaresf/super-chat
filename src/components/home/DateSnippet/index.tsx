import React from 'react'
import DateSnippet from './DateSnippet'
import formatDate from './formatDate'

interface ContainerProps {
  date: Date
}

const Container = ({ date }: ContainerProps) => (
  <DateSnippet date={formatDate(date)} />
)

export default Container
