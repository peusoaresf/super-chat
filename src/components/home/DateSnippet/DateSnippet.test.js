import React from 'react'
import renderer from 'react-test-renderer'
import DateSnippet from './DateSnippet'

test(`renders correctly`, () => {
  const tree = renderer.create(<DateSnippet date="yesterday" />).toJSON()
  expect(tree).toMatchSnapshot()
})