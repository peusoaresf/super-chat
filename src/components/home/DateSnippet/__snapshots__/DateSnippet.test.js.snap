// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    Object {
      "fontSize": 11,
      "marginBottom": 10,
    }
  }
>
  yesterday
</Text>
`;
